gems/pending/spec/spec_helper.rb

```diff
@@ -32,3 +36,30 @@
   config.backtrace_exclusion_patterns << %r{/lib\d*/ruby/[0-9]}
   config.backtrace_exclusion_patterns << %r{/gems/[0-9][^/]+/gems/}
 end
+
+#
+# So tests can clear class-level caches between examples.
+# TODO: Add this to the azure-armrest gem.
+#
+class Azure::Armrest::ArmrestService
+  def self.clear_caches
+    @@providers_hash = {}
+    @@tokens         = {}
+    @@subscriptions  = {}
+  end
+end
```

Comment:
I'm going to make a pull request for the azure-armrest gem to add this method to Azure::Armrest::ArmrestService. I don't want the release of the new version of the gem to hold up the merge of this PR, so I'm going to leave this here for now, and remove it in another PR after the gem is released.